modules.d/90nvdimm/module-setup.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# called by dracut
+check() {
+    if [[ ! $hostonly ]]; then
+        return 0
+    fi
+    [[ $DRACUT_KERNEL_MODALIASES && -f "$DRACUT_KERNEL_MODALIASES" ]] && \
+        grep -q libnvdimm "$DRACUT_KERNEL_MODALIASES" && return 0
+    return 255
+}
+
+# called by dracut
+depends() {
+    return 0
+}
+
+# called by dracut
+installkernel() {
+    # Directories to search for NVDIMM "providers" (firmware drivers)
+    # These modules call "nvdimm_bus_register()".
+    local _provider_dirs='=drivers/nvdimm =drivers/acpi =arch/powerpc'
+
+    #instmods() will take care of hostonly
+    dracut_instmods -o -s nvdimm_bus_register $_provider_dirs
+}
+
+# called by dracut
+install() {
+    inst /usr/bin/ndctl
```

Comment:
Is this for debugging purposes? What needs this in the initramfs?

Reply:
Maybe better to use:
```
inst_multiple -o ndctl
```
and have it optional and without the path.

Reply:
> Is this for debugging purposes? What needs this in the initramfs?

This is needed for encrypted NVDIMMs.

    $ cat /etc/modprobe.d/nvdimm-security.conf
    install libnvdimm /usr/bin/ndctl load-keys ; /sbin/modprobe --ignore-install libnvdimm $CMDLINE_OPTS

This is subobtimal; "install" in modprobe.d is more or less deprecated, and this "install" statement is only needed for users of encrypted NVDIMMs which probably represent a tiny minority of the already low number of NVDIMM users. OTOH, without this, booting from encrypted NVDIMM devices simply fails.

Reply:
Ok, can you change it to ```inst_multiple -o ndctl``` nevertheless?